app/src/main/java/mozilla/lockbox/presenter/IntentPresenter.kt

```diff
@@ -1,20 +1,59 @@
 package mozilla.lockbox.presenter
 
+import android.support.v4.app.Fragment
+import android.support.v4.app.FragmentManager
+import android.support.v7.app.AppCompatActivity
 import android.util.Log
+import mozilla.lockbox.R
 import mozilla.lockbox.action.RouteAction
-import mozilla.lockbox.flux.Dispatcher
+import mozilla.lockbox.action.RouteAction.*
 import mozilla.lockbox.store.RouteStore
+import mozilla.lockbox.view.FxALoginFragment
+import mozilla.lockbox.view.WelcomeFragment
 
-class IntentPresenter(routeStore: RouteStore = RouteStore.shared) {
-    companion object {
-        val shared = IntentPresenter()
-    }
+
+class IntentPresenter(private val activity: AppCompatActivity, routeStore: RouteStore = RouteStore.shared) {
+
+    val welcome: WelcomeFragment by lazy { WelcomeFragment() }
+    val login: FxALoginFragment by lazy { FxALoginFragment() }
 
     init {
         routeStore.routes.subscribe { a -> route(a) }
     }
 
+    fun onViewReady() {
+        replaceFragment(this.welcome, false)
```

Comment:
This is where we might need to worry about persisting/restoring the current route.